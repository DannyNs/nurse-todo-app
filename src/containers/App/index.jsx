import React from 'react';
import PropType from 'prop-types';
import { observer } from 'mobx-react';
import { TasksStore } from '../../stores/TaskStore';
import NewTask from '../../stores/TaskStore/NewTask';

import TasksBoard from '../../components/TasksBoard';
import NewTaskForm from '../../components/NewTaskForm';

import './index.scss';

@observer
class App extends React.Component {
  render() {
    const {
      title,
      tasksStore: {
        tasks, addNewTask,
      },
    } = this.props;

    return (
      <div className="nr-app">
        <div className="nr-app__controls">
          <h1 className="nr-app__controls--center">{title}</h1>
        </div>
        <NewTaskForm addNewTask={addNewTask} newTask={new NewTask()} />
        <hr />
        <TasksBoard tasks={tasks} />
      </div>
    );
  }
}

App.propTypes = {
  title: PropType.string.isRequired,
  tasksStore: PropType.instanceOf(TasksStore).isRequired,
};

export default App;
