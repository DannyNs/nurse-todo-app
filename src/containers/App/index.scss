.nr-app {
  width: 100%;
  height: 100%;

  &__title {
    background-color: var(--main-success-color);

    h2 {
      color: var(--light-text-color);
      font-weight: bold;
    }
  }

  &__add-new-task {
    margin: 20px;
  }
}