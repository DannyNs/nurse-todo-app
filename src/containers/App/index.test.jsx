import React from 'react';
import { shallow } from 'enzyme';
import App from '.';
import tasksStore from '../../stores/TaskStore';

describe('App', () => {
  it('should render without crashing', () => {
    // given
    const title = 'Test 123';

    // when
    const wrapper = shallow(<App
      title={title}
      tasksStore={tasksStore}
    />);

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
