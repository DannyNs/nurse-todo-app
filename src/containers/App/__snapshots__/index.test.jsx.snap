// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<div
  className="nr-app"
>
  <div
    className="nr-app__controls"
  >
    <h1
      className="nr-app__controls--center"
    >
      Test 123
    </h1>
  </div>
  <_class
    addNewTask={[Function]}
    newTask={NewTask {}}
  />
  <hr />
  <_class
    tasks={
      Array [
        Task {
          "description": "Change pillow",
          "done": true,
          "id": 1,
          "name": "Brian Adams",
          "time": 1542907030863,
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "Task@1.keys",
              "observers": Set {},
            },
            "name": "Task@1",
            "target": [Circular],
            "values": Map {
              "done" => true,
            },
          },
        },
        Task {
          "description": "Make bed",
          "done": false,
          "id": 2,
          "name": "Bob Dylan",
          "time": 1542907031866,
          Symbol(mobx did run lazy initializers): true,
          Symbol(mobx administration): ObservableObjectAdministration$$1 {
            "defaultEnhancer": [Function],
            "keysAtom": Atom$$1 {
              "diffValue": 0,
              "isBeingObserved": false,
              "isPendingUnobservation": false,
              "lastAccessedBy": 0,
              "lowestObserverState": 2,
              "name": "Task@2.keys",
              "observers": Set {},
            },
            "name": "Task@2",
            "target": [Circular],
            "values": Map {
              "done" => false,
            },
          },
        },
      ]
    }
  />
</div>
`;
